### PR TITLE
Add demo refinery template

### PR DESCRIPTION
**Pull request by bricesanchez**
This is the template to create a demo of Refinery CMS.

How could we automaticaly create the first user ?

### CODE REVIEW

File: templates/refinery/demo.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+gem 'seo_meta', git: 'https://github.com/parndt/seo_meta', branch: 'master'
+
+gem 'paper_trail', git: 'https://github.com/airblade/paper_trail', branch: 'master'
+
+gem 'refinerycms-blog', git: 'https://github.com/refinery/refinerycms-blog', branch: 'master'
+gem 'refinerycms-inquiries', git: 'https://github.com/refinery/refinerycms-inquiries', branch: 'master'
+gem 'refinerycms-news', git: 'https://github.com/refinery/refinerycms-news', branch: 'master'
+gem 'refinerycms-page-images', git: 'https://github.com/refinery/refinerycms-page-images', branch: 'master'
+gem 'refinerycms-portfolio', git: 'https://github.com/refinery/refinerycms-portfolio', branch: 'master'
+gem 'refinerycms-settings', git: 'https://github.com/refinery/refinerycms-settings', branch: 'master'
```
Comment (parndt) [inline templates/refinery/demo.rb]: do we really need settings?

**Comment by parndt** [comment]
To create the first user we just do this:

```
bin/rails runner "Refinery::User.new(:username => 'demo', :password => 'demo', :email => 'demo@example.org').create_first"
```

File: templates/refinery/demo.rb (hunk 1 above)
Comment (bricesanchez) [inline templates/refinery/demo.rb]: I think yes until this gem is up-to-date. A demo is great to test gems in order to be able to know if we really need them in a real project.

File: templates/refinery/demo.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+rescue
+  gsub_file 'Gemfile', "# gem 'therubyracer'", "gem 'therubyracer'"
+end
+
+run 'bundle install'
+
+rake 'db:create'
+generate "refinery:cms --fresh-installation #{ARGV.join(' ')}"
+
+run ' generate refinery:blog'
```
Comment (bricesanchez) [inline templates/refinery/demo.rb]: Oups i fix that in next commit

File: templates/refinery/demo.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+run 'rails generate refinery:portfolio'
+run 'rails generate refinery:settings'
+run 'rails generate refinery:search'
+
+rake 'db:migrate'
+rake 'db:seed'
+
+run "bin/rails runner \"Refinery::User.new(:username => 'demo', :password => 'demo', :email => 'demo@example.org').create_first\""
+
+say <<-SAY
```
Comment (simi) [inline templates/refinery/demo.rb]: We don't need this IMHO.

File: templates/refinery/demo.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+gem 'refinerycms-news', git: 'https://github.com/refinery/refinerycms-news', branch: 'master'
+gem 'refinerycms-page-images', git: 'https://github.com/refinery/refinerycms-page-images', branch: 'master'
+gem 'refinerycms-portfolio', git: 'https://github.com/refinery/refinerycms-portfolio', branch: 'master'
+gem 'refinerycms-settings', git: 'https://github.com/refinery/refinerycms-settings', branch: 'master'
+gem 'refinerycms-search', git: 'https://github.com/refinery/refinerycms-search', branch: 'master'
+"
+end
+
+begin
+  require 'execjs'
```
Comment (simi) [inline templates/refinery/demo.rb]: It is required already.

File: templates/refinery/demo.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  ::ExecJS::Runtimes.autodetect
+rescue
```
Comment (simi) [inline templates/refinery/demo.rb]: Catch [RuntimeUnavailable](https://github.com/sstephenson/execjs/blob/master/lib/execjs/module.rb#L8) only. This should be moved to the top begin rescue block.

**Comment by bricesanchez** [comment]
@simi is it ok for you?

**Comment by simi** [comment]
We need to test this with Phil on server once it will be ready for
automatic deployment.

2015-02-24 22:51 GMT+01:00 Brice Sanchez <notifications@github.com>:

**Comment by parndt** [comment]
@simi are we able to test this yet?

**Comment by simi** [comment]
Actually this was tested, but we need to move it to refinery office :)

**Comment by parndt** [comment]
@simi OK, should we merge it or something else?

**Comment by bricesanchez** [comment]
@simi :wave:

**Comment by simi** [comment]
damn, I need to refresh my head for this

It is working with - https://github.com/simi/refinerycms/commit/42288ed837bd226777e9c7423e45ddcae60662ff. But I doesn't keep git history fort demo repo, but I think we can live with that for now.

**Comment by bricesanchez** [comment]
@simi So, Is this PR useless? Could i close it and wait for your PR ?

**Comment by simi** [comment]
feel free to wait unless my work is done

**Pull request merged**